catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -639,7 +629,9 @@ private ResourceInfo getResourceInfo(
 
   private void validateResourceInfoRequest(String name, Object value)
       throws ResourceNotSupportedException {
-    if (!(ResourceRequest.GET_RESOURCE_BY_PRODUCT_URI.equals(name) && value instanceof URI)
+    if (!((ResourceRequest.GET_RESOURCE_BY_PRODUCT_URI.equals(name)
```

Comment:
❓ I'm curious if there is some way we can get rid of this `validateResourceInfoRequest` method. Currently we would only allow resource requests by these 3 attributes we check here. We may be able to do a catalog query based on the incoming request attribute/value and then fetching that metacards resource.